src/BidirectedGraph.cpp

```diff
@@ -1,27 +1,30 @@
 #include "BidirectedGraph.hpp"
 #include <queue>
 
-vector<uint64_t> BidirectedGraph::get_reachable_nodes(uint64_t id){
+BidirectedGraph::BidirectedGraph() {
+
+}
+
+vector<nid_t> BidirectedGraph::get_reachable_nodes(nid_t id){
     return reachable_nodes.at(id);
 }
 
-void BidirectedGraph::add_edge(uint64_t id1, uint64_t id2, bool from_left, bool to_right){
-    BidirectedEdge tempedge(id1, id2);
-    tempedge.from_left = from_left;
-    tempedge.to_right = to_right;
-  
+void BidirectedGraph::add_edge(nid_t id1, nid_t id2, bool from_left, bool to_right){
+    BidirectedEdge from_id1(id1, id2, from_left, to_right);
+    BidirectedEdge from_id2(id2, id1, to_right, from_left);
```

Comment:
Check me on this, but I think you might want `BidirectedEdge from_id2(id2, id1, !to_right, !from_left);`
If the edge was "from left", but now the "from" is the "to", then it should be "to left" i.e. "not to right".

Reply:
You are completely correct. As I was typing up an example I realized my mistake. Thanks for catching this!

Reply:
You are absolutely correct. I don't think I had the best understanding of a degenerate edge when I wrote that piece of code. I'll make that change immediately. Thanks for catching it!